File: web/src/types/storage-location.ts
export interface StorageLocation {
  id: number;
  name: string;
  base_path: string;
  type?: 'local' | 'sftp';
  address?: string;
  port?: number;
  remote_path?: string;
  username?: string;
  password?: string;
  ssh_key?: string;
  auth_type?: 'key' | 'password';
  enabled?: boolean;
  created_at: string;
}

export interface StorageLocationCreateInput {
  name: string;
  base_path?: string;
  type?: 'local' | 'sftp';
  address?: string;
  port?: number;
  remote_path?: string;
  username?: string;
  password?: string;
  ssh_key?: string;
  auth_type?: 'key' | 'password';
  enabled?: boolean;
}

File: server/entity/storage_location.go
package entity

import "time"

// StorageLocation defines where backups are stored
type StorageLocation struct {
	ID        uint      `gorm:"primaryKey" json:"id"`
	Name      string    `gorm:"not null" json:"name"`
	BasePath  string    `gorm:"not null" json:"base_path"`
	Type      string    `gorm:"default:local" json:"type"`
	Address   string    `json:"address,omitempty"`
	Port      int       `json:"port,omitempty"`
	RemotePath string   `json:"remote_path,omitempty"`
	Username  string    `json:"username,omitempty"`
	Password  string    `json:"password,omitempty"`
	SSHKey    string    `json:"ssh_key,omitempty"`
	AuthType  string    `json:"auth_type,omitempty"`
	Enabled   bool      `gorm:"default:true" json:"enabled"`
	CreatedAt time.Time `json:"created_at"`
}
